### ISSUE TITLE
Microsoft SQL Server Management Studio Editor 2014 doesn't work with NVDA 2017.2 - With Workaround

### ISSUE THREAD

**Issue by albzan** [nvaccess/nvda#7235]
When the Text Editor (Query editor) is focused, the text is not reported and an exception occurs instead.

### Steps to reproduce:
1. Open SQL Server Management Studio 2014 (also with the Express version)
2. From the File Menu select New Query
3. Type something or move using keyboard

### Expected behavior:
The text editor should be reported. It works fine with previous version of NVDA

### Actual behavior:
The text is not reported and the following error is present if the log level is set in debug/warning:
error executing script: <bound method Dynamic_EditableTextWithoutAutoSelectDetectionWpfTextViewUIA.script_caret_moveByLine of <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionWpfTextViewUIA object at 0x050745F0>> with gesture u'up arrow'
Traceback (most recent call last):
  File "scriptHandler.pyo", line 187, in executeScript
  File "editableText.pyo", line 143, in script_caret_moveByLine
  File "editableText.pyo", line 103, in _caretMovementScriptHelper
  File "editableText.pyo", line 89, in _caretScriptPostMovedHelper
  File "speech.pyo", line 689, in speakTextInfo
  File "NVDAObjects\UIA\__init__.pyo", line 584, in getTextWithFields
  File "NVDAObjects\UIA\__init__.pyo", line 574, in _getTextWithFieldsForUIARange
  File "NVDAObjects\UIA\__init__.pyo", line 412, in _getTextWithFields_text
  File "NVDAObjects\UIA\__init__.pyo", line 409, in _getTextWithFields_text
  File "NVDAObjects\UIA\__init__.pyo", line 157, in _getFormatFieldAtRange
  File "UIAUtils.pyo", line 189, in __init__
COMError: (-2147220991, 'An event was unable to invoke any of the subscribers', (None, None, None, 0, None))

### System configuration:
NVDA version: 2017.2
Microsoft SQL Server Management Studio 2014
Microsoft Windows 10 Creators Update

NVDA Installed or portable: Installer

### Other questions:
Have you tried any other versions of NVDA?
Yes, the issue does not affects previous stable versions of NVDA.

### Workaround:
Since this is in a production computer in my workplace, i developed a workaround.  You can find it attached to this message.
It also contains the workaround for the issue of the tree view already reported on #5517

[Attachment: ssms.zip](https://github.com/nvaccess/nvda/files/1040849/ssms.zip)

**Comment by Adriani90** [comment]
@albzan could you please give an update to this one? Does it work better now in NVDA 2019.1.1?

**Comment by leonardder** [comment]
This works for me in SSMS 18. As SSMS is free software, a user of an older version is able to update. Closing this issue.
